samples/news/News.UITests/Pages/FavoritesPage.cs

```diff
@@ -12,17 +13,43 @@ public class FavoritesPage : BasePage
         };
 
         protected readonly Query FavoriteButton;
+        protected readonly Query EmptyLabel;
+        protected readonly Query NewsInfiniteListView;
 
         public FavoritesPage()
         {
             FavoriteButton = x => x.Marked(nameof(FavoriteButton));
+            NewsInfiniteListView = x => x.Marked(nameof(NewsInfiniteListView));
+            EmptyLabel = x => x.Marked(nameof(EmptyLabel));
         }
 
-        public void RemoveFavorite()
+        public FavoritesPage VerifyThatEmpty()
+        {
+            app.WaitForElement(EmptyLabel);
+            app.Screenshot("Verified that favorite is empty");
+            return this;
+        }
+
+        public FavoritesPage VerifyThatNotEmpty()
+        {
+            app.WaitForNoElement(EmptyLabel);
+            app.Screenshot("Verified that favorite is not empty");
+            return this;
+        }
+
+        public FavoritesPage RefreshFavorites()
+        {
+            app.WaitForElement(NewsInfiniteListView);
+            var newsList = app.Query(NewsInfiniteListView).First();
+            app.DragCoordinates(newsList.Rect.X, newsList.Rect.Y, newsList.Rect.X, newsList.Rect.CenterY);
```

Comment:
the was the trick to make pull to refresh work, we need specific coordinates of the ui table view to know where to pull. All the rest is straightforward.

Reply:
Yup! That's it!